src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -156,6 +160,7 @@ void addConjunct(ref IsPatternTranslator self, BoundExpression expression)
                     case BoundDagFieldEvaluation f:
                         {
                             var field = f.Field;
+                            field = field.TupleUnderlyingField ?? field;
```

Comment:
>field = field.TupleUnderlyingField ?? field; [](start = 28, length = 44)

I believe simply using TupleUnderlyingField is not sufficient for correct handling of long tuples. Instead, we should probably use MakeFieldAccess helper rather than ```_factory.Field``` below. Please add a test that uses Item9 field, for example.

Reply:
Noted in https://github.com/dotnet/roslyn/pull/24059/commits/114a56e9077ed03f08443889b69cae6972a4acc6 of https://github.com/dotnet/roslyn/pull/24059